Update README.MD (#1)
mulesoft-labs/anypoint-exchange-public-portal-proxy-image: README.MD

# Exchange public portal vanity domain setup

> This repository shows the necessary nginx configuration for setting up an organization's public portal using a vanity domain. It also provides a docker image for running this configuration.

Reviewer: ```suggestion
This repository contains instructions and the necessary nginx configuration for setting up an organization's public portal using a vanity domain. It also provides a docker image for running this configuration.
```

## Whitelisting your domain

> To allow the Exchange public portal to work with your own domain, it first needs to be whitelisted. To do so:

Reviewer: ```suggestion
Your domain needs to be whitelisted in Exchange for your public portal to work properly. To do so:
```

> 1. Get an access token from an administrator user of your organization in the Anypoint Platform:

Reviewer 2: `... from a user with Organization Administrators role in the master organization`?

```
ACCESS_TOKEN=$(curl https://anypoint.mulesoft.com/accounts/login -XPOST -d "username=YOUR_USERNAME&password=YOUR_PASSWORD" | jq -r ".access_token")
```

> 2. Use that token to set your custom domain in your organization:

Reviewer: ```suggestion
2. Use that token to whitelist your custom domain:
```

```

> curl https://anypoint.mulesoft.com/exchange/api/v1/organizations/YOUR_ORGANIZATION_DOMAIN/portal/domain -XPUT -d "your.custom.domain" -H "content-type: text/plain" -H "authorization: bearer ACCESS_TOKEN"

Reviewer: ```suggestion
curl https://anypoint.mulesoft.com/exchange/api/v1/organizations/YOUR_ORGANIZATION_DOMAIN/portal/domain -XPUT -d "your.custom.domain" -H "content-type: text/plain" -H "authorization: bearer $ACCESS_TOKEN"
```

```

Keep in mind that if this step is not followed, you wont be able to log in to the portal or perform any actions that need authentication.

## Setting up your own nginx configuration

> If your organization already has a running nginx configuration, you must include the necessary rules found [here](https://github.com/mulesoft-labs/anypoint-exchange-public-portal-proxy-image/blob/master/config/nginx.conf.template#L11-L30) replacing the following variables with what may apply: 

Reviewer 2: I would just mention that is mandatory to run the server using `HTTPS`

```
  PUBLIC_PORTAL_PATH: Your custom path. Must end with /.
  ANYPOINT_BASE_URL: Anypoint platform url for the desired region.
  ORGANIZATION_DOMAIN: Your organization's domain in anypoint platform.
```

For example, for the `Robocop` organization public portal with an anypoint domain of `robo-cop` to run on `developers.robocop.com/path/of/justice/`:

```
  PUBLIC_PORTAL_PATH: /path/of/justice/
  ANYPOINT_BASE_URL: https://anypoint.mulesoft.com
  ORGANIZATION_DOMAIN: robo-cop
```

If set up correctly, you should be able to access and navigate `https://developers.robocop.com/path/of/justice/` exactly as if you were at `https://anypoint.mulesoft.com/exchange/portals/robo-cop/`.

## Running the docker image

For a minimal working setup, there is an nginx based docker image available.

1. Get the image from docker hub:

```

> docker pull joaquinstankus/anypoint-exchange-public-portal-proxy:latest

Reviewer: let's find out who's the owner of https://hub.docker.com/u/mulesoft/ and publish the image there

Reviewer 2: +1
